src/Sparrow/Json/AbstractBlittableJsonTextWriter.cs

```diff
@@ -336,29 +350,29 @@ private void UnlikelyWriteLargeString(byte* strBuffer, int size, int numberOfEsc
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void WriteEscapeCharacter(byte* buffer, byte b)
         {
+            // We don't need to call it multiple times, we can just ensure we have enough and be done with it.
+            EnsureBuffer(6);
```

Comment:
Why `6` is the number here? I see that previously we passed `6` or `2` depending on the `r` value. Can you elaborate more on this change?

Reply:
Because we are ensuring that on any sequence we can write a complete escape sequence regardless of the path we follow. In that way, we can check that once causing inlining code and improved block execution.